sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldAccessDescriptor.java

```diff
@@ -379,6 +376,13 @@ public FieldAccessDescriptor resolve(Schema schema) {
       field = fillInMissingQualifiers(field, schema);
       fields.add(field);
     }
+
+    if (!getFieldInsertionOrder()) {
+      // Re-order fields based on field ID, rather than keeping them in insertion order
+      Function<FieldDescriptor, Integer> extract =
+          (Function<FieldDescriptor, Integer> & Serializable) FieldDescriptor::getFieldId;
+      Collections.sort(fields, Comparator.comparing(extract));
```

Comment:
Does it work to simply write Collections.sort(fields, Comparator.comparing(FieldDescriptor::getFieldId))?

We needed the verbose version before because it was captured inside the Set class, so had to be serializable. I suspect we can drop back to the simple version now.

Reply:
Good point, that does seem to work now :+1: